hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1385,6 +1387,46 @@ def minimize_nodes(self, nodes):
         else:
             raise NotImplementedError
 
+    @defines_shrink_pass()
+    def zero_examples(self, chooser):
+        i = chooser.choose(range(len(self.examples)))
+
+        def trivial_value(ir_type):
+            # TODO: In an ideal world this would take into account the
+            # node kwargs, but this is an adequate first approximation.
+            return {
+                "integer": 0,
+                "string": "",
+                "boolean": False,
+                "float": 0.0,
+                "bytes": b"",
+            }[ir_type]
+
+        prev = self.shrink_target
+        nodes = self.shrink_target.ir_nodes
+        ex = self.examples[i]
+        prefix = nodes[: ex.ir_start]
+        replacement = tuple(
+            [
+                (
+                    node
+                    if node.was_forced
+                    else node.copy(with_value=trivial_value(node.ir_type))
+                )
+                for node in nodes[ex.ir_start : ex.ir_end]
+            ]
+        )
+        suffix = nodes[ex.ir_end :]
+        attempt = self.cached_test_function_ir(prefix + replacement + suffix)
+
+        if self.shrink_target is not prev:
+            return
+
+        if attempt is not None and not isinstance(attempt, _Overrun):
+            new_ex = attempt.examples[i]
+            new_replacement = attempt.ir_nodes[new_ex.start : new_ex.end]
```

Comment:
```suggestion
            new_replacement = attempt.ir_nodes[new_ex.ir_start : new_ex.ir_end]
```

I'm surprised this didn't cause index errors in tests, since start/end indicate the buffer index, which is likely to be much larger than the node index.